test/unit/models/users.test.js

```diff
@@ -115,6 +115,16 @@ describe("users", function () {
       const { user } = await users.fetchUser({ githubUsername });
       expect(user).to.haveOwnProperty("github_created_at");
     });
+
+    it("it should filter out the id field while updating profileDiff", async function () {
+      const userData = userDataArray[0];
+      const { userId } = await users.addOrUpdate(userData);
+      const profileDiff = { id: "random-id", diff: "random-diff" };
+      await users.addOrUpdate(profileDiff, userId);
```

Comment:
we should verify how overlapping keys between user.data() and profileDiff are resolved.

Reply:
I think this had to be resolved because we are removing PR #2282. We had a discussion around this.